### PR TITLE
feat: improve relay modal 

### PR DESCRIPTION
**Pull request by 0xArdy**

### CODE REVIEW

File: src/containers/Modals/RelayModal/CallerSection.tsx [hunk 1]
```diff
@@ -16,71 +16,45 @@ interface CallerSectionProps {
...
-    setCallers([]);
-    setCallerAddress('');
-    setAllowAnyCaller(!allowAnyCaller);
-  };
-
-  const handleAddNewCaller = () => {
-    if (isAddress(callerAddress)) {
-      setCallers([callerAddress, ...callers]);
+    const newValue = !allowAnyCaller;
+    if (newValue) {
```
Comment (turtlemoji) [inline src/containers/Modals/RelayModal/CallerSection.tsx:36]: the const declaration maybe is unneeded

File: src/containers/Modals/RelayModal/CallerSection.tsx [hunk 2]
```diff
@@ -92,57 +66,48 @@ export const CallerSection = ({ callersList, setCallersList, isLoading, setIsErr
...
-        </TextButton>
-
         <Container>
           <SSwitch disabled={isLoading} onClick={handleToggle} />
           <ToggleText>Allow any caller</ToggleText>
         </Container>
       </CallersContainer>
+
+      {!allowAnyCaller && !!callersList.length && (
+        <>
```
Comment (turtlemoji) [inline src/containers/Modals/RelayModal/CallerSection.tsx:97]: needed?

File: src/containers/Modals/RelayModal/JobSection.tsx [hunk 3]
```diff
@@ -2,181 +2,211 @@ import { Box, Radio, styled } from '@mui/material';
...
-          disabled
+          placeholder='Enter function selector...'
+          value={functionSelector}
+          setValue={setFunctionSelector}
+          onClick={addNewSelector}
+          disabled={isLoading}
           dataTestId='function-selector-input'
+          error={selectorRepeated}
+          errorText={errorText}
+          addable={!!functionSelector && !isLoading && isHex(functionSelector) && functionSelector.length === 10}
```
Comment (turtlemoji) [inline src/containers/Modals/RelayModal/JobSection.tsx:176]: I would create a `isAddable` var for this.

File: src/hooks/useAbi.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+export const useAbi = () => {
+  const [abi, setAbiData] = useState<{ [key: string]: string }>({});
+  const { currentNetwork } = useStateContext();
+  const { ETHERSCAN_KEY } = getConfig();
+
+  const aliasKey = 'xkeeper-abis';
+
+  const getAbi = useCallback(
+    async (contractAddress: string) => {
+      console.log('getAbi', contractAddress);
```
Comment (turtlemoji) [inline src/hooks/useAbi.tsx:17]: remove later

**Review by turtlemoji — COMMENTED** [review]
lgtm! left some comments

File: src/containers/Modals/RelayModal/CallerSection.tsx (hunk 2 above)
Comment (0xArdy) [reply src/containers/Modals/RelayModal/CallerSection.tsx:97]: in this case yes

File: src/containers/Modals/RelayModal/CallerSection.tsx (hunk 1 above)
Comment (0xArdy) [reply src/containers/Modals/RelayModal/CallerSection.tsx]: removed https://github.com/defi-wonderland/xkeeper-app/pull/24/commits/d3d2cb391f26a13163d80ecac52b3095a3822e69

File: src/containers/Modals/RelayModal/JobSection.tsx (hunk 3 above)
Comment (0xArdy) [reply src/containers/Modals/RelayModal/JobSection.tsx]: done https://github.com/defi-wonderland/xkeeper-app/pull/24/commits/03ce99897a82904dd88edf8975c84dfbe5c921ec

File: src/hooks/useAbi.tsx (hunk 4 above)
Comment (0xArdy) [reply src/hooks/useAbi.tsx]: fixed https://github.com/defi-wonderland/xkeeper-app/pull/24/commits/3dd27130d3362b299cfda9711d9d0a0de6e519e8

**Review by turtlemoji — APPROVED** [review]

**Pull request merged**